### PR TITLE
feat(application-system): new data providers juni

### PR DESCRIPTION
**Pull request by alexdiljar**
Updating data providers for Júní CODEOWNERS applications. 
The breakdown of the changes can be viewed here. https://www.notion.so/New-data-providers-J-n-d70ce1b98a58496590b4beb1a576cea2

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/template-api-modules/src/lib/modules/templates/driving-license-submission/driving-license-submission.service.ts [hunk 1]
```diff
@@ -157,24 +159,37 @@ export class DrivingLicenseSubmissionService extends BaseTemplateApiService {
...
+              err.problem?.title || coreErrorMessages.failedDataProviderSubmit,
+            summary: err.problem?.detail || '',
+          },
+          400,
+        )
+      }
+    }
     return {
-      success: result.success,
+      success: false,
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/driving-license-submission/driving-license-submission.service.ts:192]:
```javascript
return {
  success: false
}
```

at the function top level, after a try block wherein the try part returns nothing.
Does this not result in success always being reported as false or am I missing something?
Is this intentional perhaps?

File: libs/application/template-api-modules/src/lib/modules/templates/p-sign-submission/p-sign-submission.service.ts [hunk 2]
```diff
@@ -64,6 +66,24 @@ export class PSignSubmissionService extends BaseTemplateApiService {
     this.s3 = new S3()
   }
 
+  async doctorsNote({
+    auth,
+  }: TemplateApiModuleActionProps): Promise<CertificateInfoResponse> {
+    console.log('HELLO')
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/p-sign-submission/p-sign-submission.service.ts:72]: Hello :wave: :smile_cat:

File: libs/application/types/src/lib/template-api/shared-api/shared-api-definitions/driving-license.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { defineTemplateApi } from '../../TemplateApi'
+
+const namespace = 'DrivingLicenseShared'
+
+export const HasTechingRightsApi = defineTemplateApi({
```
[the remark this answers is not in the document]
Comment (kksteini) [reply libs/application/types/src/lib/template-api/shared-api/shared-api-definitions/driving-license.ts:5]: Although maybe this is spread as far as Juristiction. Just hoping to catch a typo early :sweat_smile:

File: libs/application/templates/driving-license/src/lib/drivingLicenseTemplate.ts [hunk 4]
```diff
@@ -63,6 +73,26 @@ const template: ApplicationTemplate<
               },
               write: 'all',
               delete: true,
+              api: [
+                NationalRegistryUserApi,
+                TeachersApi,
+                UserProfileApi,
+                PaymentCatalogApi.configure({
+                  params: { orginizationId: '6509142520' },
```
Comment (kksteini) [inline libs/application/templates/driving-license/src/lib/drivingLicenseTemplate.ts:81]: `libs/application/templates/passport/src/lib/PassportTemplate.ts` stores the organization id in constants where it is easier to find and change.
Consider doing the same here?

File: libs/application/templates/operating-license/src/lib/OperatingLicenseTemplate.ts [hunk 5]
```diff
@@ -59,6 +61,13 @@ const OperatingLicenseTemplate: ApplicationTemplate<
               ],
               write: 'all',
               delete: true,
+              api: [
+                PaymentCatalogApi.configure({
+                  params: { orginizationId: '6509142520' },
```
Comment (kksteini) [inline libs/application/templates/operating-license/src/lib/OperatingLicenseTemplate.ts:66]: `libs/application/templates/passport/src/lib/PassportTemplate.ts` stores the organization id in constants where it is easier to find and change.
Consider doing the same here?

File: libs/application/template-api-modules/src/lib/modules/shared/api/driving-license/driving-license.service.ts [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    }
+    const hasQualityPhoto = await this.drivingLicenseService.getHasQualityPhoto(
+      { nationalId: auth.nationalId },
+    )
+    return {
+      hasQualityPhoto,
+    }
+  }
+
+  async juristictions(): Promise<Juristiction[]> {
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/api/driving-license/driving-license.service.ts:157]: jurisdiction

File: libs/application/template-api-modules/src/lib/modules/shared/api/driving-license/driving-license.service.ts [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+    application,
+  }: TemplateApiModuleActionProps): Promise<HasQualityPhoto> {
+    // If running locally or on dev allow for fake data
+    const useFakeData = getValueViaPath<'yes' | 'no'>(
+      application.answers,
+      'fakeData.useFakeData',
+    )
+
+    if (useFakeData === 'yes') {
+      const hasQualityPhoto = getValueViaPath<'yes' | 'no'>(
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/api/driving-license/driving-license.service.ts:143]: These 'yes' 'no' could be moved to a shared const in types maybe?

File: libs/application/template-api-modules/src/lib/modules/shared/api/driving-license/sortTeachers.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import { Teacher } from '@island.is/clients/driving-license'
+
+const order =
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/api/driving-license/sortTeachers.ts:3]: This something that could be utilized elsewhere in the monorepo as a shared utility?

File: libs/application/template-api-modules/src/lib/modules/shared/api/driving-license/types.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+}
+export type HasQualitySignature = {
+  hasQualitySignature: boolean
+}
+
+export const YES = 'yes'
+export const NO = 'no'
+
+type FakeCurrentLicense = 'none' | 'temp'
+type YesOrNo = 'yes' | 'no'
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/api/driving-license/types.ts:12]: Yep this one from above

**Review by obmagnusson — APPROVED** [review]
🎈 💯  👍 🎉 .Awesome! Couple of tiny comments. Think we should just merge and start testing on a feature deploy as soon as possible.

**Pull request merged**